tests/many_to_one/tests.py

```diff
@@ -522,6 +522,27 @@ def test_fk_assignment_and_related_object_cache(self):
         self.assertIsNot(c.parent, p)
         self.assertEqual(c.parent, p)
 
+    def test_assign_fk_id_none(self):
+        parent = Parent.objects.create(name='jeff')
+        child = Child.objects.create(name='frank', parent=parent)
+        parent.bestchild = child
+        parent.save()
+        parent.bestchild_id = None
+        parent.save()
+        self.assertIsNone(parent.bestchild_id)
+        self.assertIsNone(parent.bestchild)
+
+    def test_assign_fk_id_value(self):
+        parent = Parent.objects.create(name='jeff')
+        child1 = Child.objects.create(name='frank', parent=parent)
+        child2 = Child.objects.create(name='randy', parent=parent)
+        parent.bestchild = child1
+        parent.save()
+        parent.bestchild_id = child2.pk
+        parent.save()
+        self.assertEqual(parent.bestchild_id, child2.pk)
+        self.assertEqual(parent.bestchild, child2)
```

Comment:
Is it worth adding a regression tests for `OneToOneField` as well as `ForeignObject` since they are also both affected by this change?